app/frontend/src/features/profile/view/leaveReference/formSteps/submit/SubmitReference.tsx

```diff
@@ -0,0 +1,134 @@
+import ReferenceOverview from "features/profile/view/leaveReference/formSteps/submit/ReferenceOverview";
+import {
+  ReferenceContextFormData,
+  useReferenceStyles,
+} from "features/profile/view/leaveReference/ReferenceForm";
+import { User } from "pb/api_pb";
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useParams } from "react-router";
+import { referenceTypeRoute } from "routes";
+import {
+  WriteFriendReferenceInput,
+  WriteHostRequestReferenceInput,
+} from "service/references";
+
+export interface SubmitReferenceProps {
+  user: User.AsObject;
+  referenceData: ReferenceContextFormData;
+}
+
+export default function SubmitReference({
+  user,
+  referenceData,
+}: SubmitReferenceProps) {
+  const {
+    writeFriendReference,
+    status: friendReferenceWritingStatus,
+    reset: resetFriendReferenceWriting,
+  } = useWriteFriendReference(user.userId);
+  const {
+    writeHostRequestReference,
+    status: hostRequestReferenceWritingStatus,
+    reset: resetHostRequestReferenceWriting,
+  } = useWriteHostReference(user.userId);
+
+  const { referenceType, hostRequest } = useParams<{
+    referenceType: string;
+    hostRequest?: string;
+  }>();
+  const theme = useTheme();
+  const classes = useReferenceStyles();
+  const isSmOrWider = useMediaQuery(theme.breakpoints.up("sm"));
+  const { handleSubmit } = useForm<ReferenceContextFormData>();
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const onFriendReferenceSubmit = () => {
+    const formData: WriteFriendReferenceInput =
+      referenceData.wasAppropriate === "true"
+        ? {
+            toUserId: user.userId,
+            wasAppropriate: true,
+            text: referenceData.text,
+            rating: referenceData.rating,
+          }
+        : {
+            toUserId: user.userId,
+            wasAppropriate: false,
+            text: referenceData.text,
+            rating: referenceData.rating,
+          };
+
+    resetFriendReferenceWriting();
+    writeFriendReference({
+      referenceData: formData,
+    });
+    window.scroll({ top: 0 });
+  };
+
+  const onHostReferenceSubmit = () => {
+    if (hostRequest && +hostRequest !== undefined) {
+      const formData: WriteHostRequestReferenceInput =
+        referenceData.wasAppropriate === "true"
+          ? {
+              hostRequestId: +hostRequest,
+              wasAppropriate: true,
+              text: referenceData.text,
+              rating: referenceData.rating,
+            }
+          : {
+              hostRequestId: +hostRequest,
+              wasAppropriate: false,
+              text: referenceData.text,
+              rating: referenceData.rating,
+            };
+
+      resetHostRequestReferenceWriting();
+      writeHostRequestReference({
+        referenceData: formData,
+      });
+      window.scroll({ top: 0 });
+    } else {
+      setErrorMessage("Error fetching the host request.");
```

Comment:
It's impossible to reach this route without the host request in the url, right? Then this isn't needed.